Compilation error: ‘numeric_limits` is not a member of ‘std’ on GCC11

System Information
-----------------------------
MultiMC version: git commit 8eee4e3e6ce1d7075ec3eae67aa9780e67d161ed

Operating System: Fedora 34 PPC64LE
Qt Creator 4.14.0
Qt 5.15.2 
GCC 11.0.0 20210123 (Red Hat 11.0.0-0), 64 bit, ppc64le

Summary of the issue or suggestion:
----------------------------------------------

Compilation yields error:

```
$ make
[  1%] Automatic MOC for target MultiMC_nbt++
[  1%] Built target MultiMC_nbt++_autogen
[  3%] Built target MultiMC_nbt++
[  3%] Automatic MOC for target systeminfo
[  3%] Built target systeminfo_autogen
[  4%] Building CXX object libraries/systeminfo/CMakeFiles/systeminfo.dir/src/sys_unix.cpp.o
/home/tle/src/MultiMC5/libraries/systeminfo/src/sys_unix.cpp: In function ‘uint64_t Sys::getSystemRam()’:
/home/tle/src/MultiMC5/libraries/systeminfo/src/sys_unix.cpp:37:26: error: ‘numeric_limits’ is not a member of ‘std’
   37 |         file.ignore(std::numeric_limits<std::streamsize>::max(), '\n');
      |                          ^~~~~~~~~~~~~~
/home/tle/src/MultiMC5/libraries/systeminfo/src/sys_unix.cpp:37:56: error: expected primary-expression before ‘>’ token
   37 |         file.ignore(std::numeric_limits<std::streamsize>::max(), '\n');
      |                                                        ^
/home/tle/src/MultiMC5/libraries/systeminfo/src/sys_unix.cpp:37:59: error: ‘::max’ has not been declared; did you mean ‘std::max’?
   37 |         file.ignore(std::numeric_limits<std::streamsize>::max(), '\n');
      |                                                           ^~~
      |                                                           std::max
In file included from /usr/include/c++/11/algorithm:62,
                 from /usr/include/qt5/QtCore/qglobal.h:142,
                 from /usr/include/qt5/QtCore/qchar.h:43,
                 from /usr/include/qt5/QtCore/qstring.h:49,
                 from /usr/include/qt5/QtCore/QString:1,
                 from /home/tle/src/MultiMC5/libraries/systeminfo/include/sys.h:2,
                 from /home/tle/src/MultiMC5/libraries/systeminfo/src/sys_unix.cpp:1:
/usr/include/c++/11/bits/stl_algo.h:3467:5: note: ‘std::max’ declared here
 3467 |     max(initializer_list<_Tp> __l, _Compare __comp)
      |     ^~~
make[2]: *** [libraries/systeminfo/CMakeFiles/systeminfo.dir/build.make:108: libraries/systeminfo/CMakeFiles/systeminfo.dir/src/sys_unix.cpp.o] Error 1
make[1]: *** [CMakeFiles/Makefile2:683: libraries/systeminfo/CMakeFiles/systeminfo.dir/all] Error 2
make: *** [Makefile:160: all] Error 2
```

What should happen:
------------------------------

No error

Suspected cause:
---------------------------

It's likely because of GCC 11: https://www.gnu.org/software/gcc/gcc-11/porting_to.html

> Header dependency changes
> Some C++ Standard Library headers have been changed to no longer include other headers that they do need to depend on. As such, C++ programs that used standard library components without including the right headers will no longer compile. The following headers are used less widely in libstdc++ and may need to be included explicitly when compiled with GCC 11:
<limits> (for std::numeric_limits)
<memory> (for std::unique_ptr, std::shared_ptr etc.)
<utility> (for std::pair, std::tuple_size, std::index_sequence etc.)

Here is how I think it should be fixed:

```diff
diff --git a/libraries/systeminfo/src/sys_unix.cpp b/libraries/systeminfo/src/sys_unix.cpp
index ab3f302e..42c0d319 100644
--- a/libraries/systeminfo/src/sys_unix.cpp
+++ b/libraries/systeminfo/src/sys_unix.cpp
@@ -4,6 +4,7 @@
 
 #include <sys/utsname.h>
 #include <fstream>
+#include <limits>
```

--- a/libraries/systeminfo/src/sys_unix.cpp
+++ b/libraries/systeminfo/src/sys_unix.cpp
@@ -4,6 +4,7 @@
 
 #include <sys/utsname.h>
 #include <fstream>
+#include <limits>
 
 Sys::KernelInfo Sys::getKernelInfo()
 {

